File: app/src/main/java/com/zhan/rippleview/MainActivity.java
package com.zhan.rippleview;

import android.animation.Animator;
import android.animation.ValueAnimator;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import zhan.rippleview.RippleView;

public class MainActivity extends AppCompatActivity implements
        RippleView.RippleAnimationListener {

  private RippleView mRippleView;
  private TextView mTextView;

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    initView();
    initListener();
  }

  private void initListener() {
    mRippleView.setRippleStateListener(this);
  }

  private void initView() {
    mRippleView = (RippleView) findViewById(R.id.root_rv);
    mTextView = (TextView) findViewById(R.id.root_tv);
  }

  public void start(View v) {
    mRippleView.startRipple();
  }

  public void stop(View v) {
    mRippleView.stopRipple();
  }

  @Override public void onAnimationUpdate(ValueAnimator animation) {
    float fraction = animation.getAnimatedFraction();
    int value = (int) (fraction * 100);
    mTextView.setText(String.valueOf(value) + "%");
  }

  @Override public void onAnimationStart(Animator animation) {
    mTextView.setText("0%");
  }

  @Override public void onAnimationEnd(Animator animation) {

  }

  @Override public void onAnimationCancel(Animator animation) {

  }

  @Override public void onAnimationRepeat(Animator animation) {

  }
}

File: app/src/main/java/com/zhan/rippleview/MainActivity.kt
package com.zhan.rippleview

import android.animation.Animator
import android.animation.ValueAnimator
import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import zhan.rippleview.RippleView

class MainActivity : AppCompatActivity(), RippleView.RippleAnimationListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initListener()
    }

    private fun initListener() {
        rippleView.setRippleStateListener(this)
    }

    fun start(view: View) {
        rippleView.startRipple()
    }

    fun stop(view: View) {
        rippleView.stopRipple()
    }

    @SuppressLint("SetTextI18n")
    override fun onAnimationUpdate(animation: ValueAnimator) {
        val fraction = animation.animatedFraction
        val value = (fraction * 100).toInt()
        root_tv.text = "$value%"
    }

    override fun onAnimationStart(animation: Animator) {
        root_tv.text = "0%"
    }

    override fun onAnimationEnd(animation: Animator) {
        // do nothing
    }

    override fun onAnimationCancel(animation: Animator) {
        // do nothing
    }

    override fun onAnimationRepeat(animation: Animator) {
        // do nothing
    }
}
